### PR TITLE
Login controller

### PR DESCRIPTION
**Pull request by Fatmasiam**
### Related:
related to #39 

- query ``` checkUsername ``` for add new saved-offer 
- route  ```/api/v1/login ```  with method post 
- integration test for the route with its query.

### Summary:
how to test this PR: 
 -  ```git fetch origin login-controller ```
 -  ```git checkout  login-controller ```
 -  ```git pull origin  login-controller```
 - ```npm run server```
 - on postman run this route ``` `/api/v1/login  ``` with method ` post ` 
 and body    :
```javascript
{
   "username":"ashatat",
    "pass":"123456"
}
```

### Checklist Criteria
- [x] **Branch is up to date with master**
- [x] **The project runs**
- - [x] You can download and install locally, and run the project locally
- - [x] All existing and new features run without bugs
- - [x] The build script runs without error
- [x] **Code is consistent**
- - [x] Linter is used, no error messages
- - [x] Code is consistent with agreed team style
- - [x] Naming of files is consistent with project (e.g. all files named with camelCase)
- [x] **No console logs**
- - [x] Console.log is only used for crucial system messages (e.g. which port server is on).
- - [x] Any console.logs used for debugging have been removed.

### CODE REVIEW

File: server/controllers/authentication/login.js [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+const bcrypt = require('bcryptjs');
+const { checkUsername } = require('../../database/queries/authentication/checkData');
+const { loginSchema } = require('../../helpers/validation-schema');
+
+module.exports = (req, res, next) => {
+  const memberInfo = { ...req.body };
+  const { error } = joi.validate(memberInfo, loginSchema);
+  if (!error) {
+    checkUsername(memberInfo.username)
+      .then(({ rows: member }) => {
```
Comment (tomduggan85) [inline server/controllers/authentication/login.js]: @Fatmasiam I have a very minor suggestion: If you are sick of typing `member[0]` you can use brackets to destructure more in this line, like this: `.then(({ rows: [ member ] }) => {`
Then below, you can just write (for example) `member.password`

File: server/controllers/authentication/authentication.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
+const { verify } = require('jsonwebtoken');
+
 module.exports = (req, res, next) => {
-  res.send('Auth');
+  const { jwt } = req.cookies;
+  const secret = process.env.SECRET;
+  if (jwt && secret) {
+    verify(jwt, secret, (err, decoded) => {
+      if (decoded) req.auth = decoded;
```
Comment (ashatat) [inline server/controllers/authentication/authentication.js]:
```suggestion
      if (decoded) req.user = decoded;
```
req.user does make more sense?

File: test/addHiredMember.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ test('add hired member test', (t) => {
       status: 'accepted',
     })
     .expect('content-type', /json/)
+    .set('Cookie', ['jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwidXNlcm5hbWUiOiJhc2hhdGF0IiwiYXZhdGFyIjoiaHR0cHM6Ly9hdmF0YXJzMS5naXRodWJ1c2VyY29udGVudC5jb20vdS8xODE0OTQzOD9zPTQ2MCZ2PTQiLCJpYXQiOjE1NjAzMjQwMzd9.zHBY7N7vfbcNiybjvJkgZY-auGUGlUEJeq4JTUlnA50'])
```
Comment (ashatat) [inline test/addHiredMember.js]: check [this](https://github.com/FACG6/logicteca/pull/182#discussion_r279756516)

File: server/database/queries/authentication/checkData.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const dbconnection = require('../../config/db_connection');
```
Comment (ashatat) [inline server/database/queries/authentication/checkData.js]: `getUser.js` make more sense I think!!!

**Pull request merged**